test/liblll/Parser.cpp

```diff
@@ -180,6 +181,290 @@ BOOST_AUTO_TEST_CASE(macro_with_zero_args)
+    "{ (CODECOPY 0x1 0x0 (CODESIZE)) [[ 0 ]] (MLOAD 0x1) (RETURN 0x0 0x0) }",
+    "{ (GASPRICE) (RETURN 0x0 0x0) }",
+    "{ (EXTCODESIZE 0x0) (RETURN 0x0 0x0)}",
+    "{ (EXTCODECOPY 0x1000000000000000000000000000000000000010 0 0 20) (RETURN 0x0 0x0) }",
+    "{ (RETURNDATASIZE) (RETURN 0x0 0x0) }",
+    "{ (RETURNDATACOPY 0x0 0x0 0x0) (RETURN 0x0 0x0) }",
+    "{ 0x00 (POP 0x00) (RETURN 0x0 0x0) }",
+    "{ (MLOAD 0x0) (RETURN 0x0 0x0)}",
+    "{ (MSTORE 0x00 0x00) (RETURN 0x0 0x0) }",
+    "{ (MSTORE8 0 0xff) (RETURN 0x0 0x0) }",
+    "{ (SLOAD 0x0) (RETURN 0x0 0x0) }",
+    "{ (SSTORE 0x1 0x1) (RETURN 0x0 0x0) }",
+    "{ (JUMP 0x4) (STOP) (JUMPDEST) (RETURN 0x0 0x0) }",
+    "{ (JUMPI 0x6 0x1) (STOP) (JUMPDEST) (RETURN 0x0 0x0) }",
+    "{ (PC) (RETURN 0x0 0x0) }",
+    "{ (MSIZE) (RETURN 0x0 0x0) }",
+    "{ (GAS) (RETURN 0x0 0x0) }",
+    "{ (JUMPDEST) (RETURN 0x0 0x0) }",
+    "{ 0xff (RETURN 0x0 0x0)}",
+    "{ 0xffff (RETURN 0x0 0x0)}",
+    "{ 0xffffff (RETURN 0x0 0x0)}",
+    "{ 0xffffffff (RETURN 0x0 0x0)}",
+    "{ 0xffffffffff (RETURN 0x0 0x0)}",
+    "{ 0xffffffffffff (RETURN 0x0 0x0)}",
+    "{ 0xffffffffffffff (return 0x0 0x0)}",
+    "{ 0xffffffffffffffff (return 0x0 0x0)}",
+    "{ 0xffffffffffffffffff (return 0x0 0x0)}",
+    "{ 0xffffffffffffffffffff (return 0x0 0x0)}",
+    "{ 0xffffffffffffffffffffff (return 0x0 0x0)}",
+    "{ 0xffffffffffffffffffffffff (return 0x0 0x0)}",
+    "{ 0xffffffffffffffffffffffffff (return 0x0 0x0)}",
+    "{ 0xffffffffffffffffffffffffffff (return 0x0 0x0)}",
+    "{ 0xffffffffffffffffffffffffffffff (return 0x0 0x0)}",
+    "{ 0xffffffffffffffffffffffffffffffff (return 0x0 0x0)}",
+    "{ 0xffffffffffffffffffffffffffffffffff (return 0x0 0x0)}",
+    "{ 0xffffffffffffffffffffffffffffffffffff (return 0x0 0x0)}",
+    "{ 0xffffffffffffffffffffffffffffffffffffff (return 0x0 0x0)}",
+    "{ 0xffffffffffffffffffffffffffffffffffffffff (return 0x0 0x0)}",
+    "{ 0xffffffffffffffffffffffffffffffffffffffffff (return 0x0 0x0)}",
+    "{ 0xffffffffffffffffffffffffffffffffffffffffffff (return 0x0 0x0)}",
+    "{ 0xffffffffffffffffffffffffffffffffffffffffffffff (return 0x0 0x0)}",
+    "{ 0xffffffffffffffffffffffffffffffffffffffffffffffff (return 0x0 0x0)}",
+    "{ 0xffffffffffffffffffffffffffffffffffffffffffffffffff (return 0x0 0x0)}",
+    "{ 0xffffffffffffffffffffffffffffffffffffffffffffffffffff (return 0x0 0x0)}",
+    "{ 0xffffffffffffffffffffffffffffffffffffffffffffffffffffff (return 0x0 0x0)}",
+    "{ 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffff (return 0x0 0x0)}",
+    "{ 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffff (return 0x0 0x0)}",
+    "{ 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff (return 0x0 0x0)}",
+    "{ 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff (return 0x0 0x0)}",
+    "{ 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff (return 0x0 0x0)}",
+    "{ (DUP1 0xff) (return 0x0 0x0)}",
```

Comment:
What is the point of the `0xff` parameter for DUPs and SWAPs? They do not take parameters and this case (for some reason) gets compiled as pushes.

Reply:
Well to be honest, `DUP` and `SWAP` make no sense for the *instructional form* in LLL.

All these are also available as inline assembly:
```
(asm 1 2 3 DUP2 SWAP3)
```

Reply:
Fixed this in #3005.